keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/Hamiltonian.java

```diff
@@ -96,15 +118,72 @@ public NetworkSamples getPosteriorSamples(final BayesianNetwork bayesNet,
 
         final Map<VertexId, ?> sampleBeforeLeapfrog = new HashMap<>();
 
-        for (int sampleNum = 1; sampleNum < sampleCount; sampleNum++) {
+        return new Sampler(latentVertices, position, gradient, momentum, positionBeforeLeapfrog,
+            gradientBeforeLeapfrog, momentumBeforeLeapfrog, random, fromVertices, leapFrogCount,
+            stepSize, logProbGradientCalculator, sampleBeforeLeapfrog, bayesNet, logOfMasterPBeforeLeapfrog
+        );
+    }
+
+    public static class Sampler implements SamplingAlgorithm {
```

Comment:
Is there anything more descriptive that this can be called other than Sampler? It seems odd that the class name here is less descriptive than the interface it implements.